specification/app/resource-manager/Microsoft.App/preview/2023-08-01-preview/Builds.json

```diff
@@ -247,6 +247,51 @@
         },
         "x-ms-long-running-operation": true
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/builders/{builderName}/builds/{buildName}/getAuthToken": {
```

Comment:
>getAuthToken [](http://example.com/codeflow?start=138&length=12)

Consider renaming this to prefix with list*. This is to satisfy a quirk in ARM templates that it only allows you to call a POST action if it is prefixed with list* . Even if get makes more sense as the naming

the examples for this are here [https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-resource#list](https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-resource#list)

You can also search for other templates online where customers use somelike listkeys on a storage account and pass the result that into other resources. By naming it with get* you block the customers from doing that.

Reply:
I updated this to be consistent with the rest of the Microsoft.App POST actions, such as [`containerapp/getAuthToken`](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/app/resource-manager/Microsoft.App/preview/2023-05-02-preview/ContainerApps.json#L1335-L1364). But yes, this makes sense to me so i will go ahead and change the naming here.